src/services/StructuralMetadataUtils.js

```diff
@@ -519,57 +548,14 @@ export default class StructuralMetadataUtils {
       });
     });
 
-    return sortedHeadings;
-  }
-
-  /**
-   * Find valid headings when either wrapping timespan before or after is null
-   * @param {Object} wrapperSpans - spans wrapping the current active timespan
-   * @param {Array} allItems - all the items in structured metadata
-   */
-  getValidHeadingForEmptySpans(wrapperSpans, allItems) {
-    let adjacentDiv = null;
-
-    let getWrapperDiv = (currentParent, position) => {
-      let wrapperParents = this.findWrapperHeaders(currentParent, allItems);
-      switch (position) {
-        case 'before':
-          return !wrapperParents.before ? currentParent : wrapperParents.before;
-        case 'after':
-          return !wrapperParents.after ? currentParent : wrapperParents.after;
-        default:
-          return currentParent;
-      }
-    };
-
-    let nestedHeadings = [];
-    let getNestedDivs = (currentHeader, currentParent) => {
-      if (currentHeader !== currentParent) {
-        let items = currentHeader.items;
-        if (items) {
-          for (let item of items) {
-            if (item.type === 'div') {
-              let { items, ...cloneWOItems } = item;
-              nestedHeadings.push(cloneWOItems);
-            }
-            getNestedDivs(item, currentParent);
-          }
-        }
+    // Filter the duplicated headings
+    sortedHeadings.filter(heading => {
```

Comment:
Just wondering if `filter` is actually filtering here, or whether it's just iterating over the array?  No big deal...

Reply:
Yeah, just iterating. I will change it to a `map`. Initially I used the `filter` , but for some reason the comparing indexes did not work for objects.